openpype/hosts/maya/plugins/publish/validate_plugin_path_attributes.py

```diff
@@ -0,0 +1,50 @@
+import os
+
+from maya import cmds
+
+import pyblish.api
+
+from openpype.pipeline.publish import ValidateContentsOrder
+
+
+class ValidatePluginPathAttributes(pyblish.api.InstancePlugin):
+    """
+    Validate plug-in path attributes point to existing file paths.
+    """
+
+    order = ValidateContentsOrder
+    hosts = ['maya']
+    families = ["workfile"]
+    label = "Plug-in Path Attributes"
+
+    def get_invalid(self, instance):
+        invalid = list()
+
+        # get the project setting
+        validate_path = (
+            instance.context.data["project_settings"]["maya"]["publish"]
+        )
+        file_attr = validate_path["ValidatePathForPlugin"]["attribute"]
+        if file_attr:
+            # get the nodes and file attributes
+            for node, attr in file_attr.items():
+                # check the related nodes
+                targets = cmds.ls(type=node)
+
+                for target in targets:
+                    # get the filepath
+                    file_attr = "{}.{}".format(target, attr)
+                    filepath = cmds.getAttr(file_attr)
+
+                    if filepath and not os.path.exists(filepath):
+                        self.log.error("File {0} not exists".format(filepath)) # noqa
+                        invalid.append(target)
```

Comment:
A would use here the guard statement to ease up on nesting and making the code more readable:

```suggestion
        file_attr = validate_path["ValidatePathForPlugin"]["attribute"]
        if not file_attr:
            return invalid
        # get the nodes and file attributes
        for node, attr in file_attr.items():
            # check the related nodes
            targets = cmds.ls(type=node)

            for target in targets:
                # get the filepath
                file_attr = "{}.{}".format(target, attr)
                filepath = cmds.getAttr(file_attr)

                if filepath and not os.path.exists(filepath):
                    self.log.error("File {0} not exists".format(filepath)) # noqa
                    invalid.append(target)
```